types/chrome/index.d.ts

```diff
@@ -9774,3 +9775,546 @@ declare namespace chrome.windows {
+        /** Rules to add. */
+        addRules?: Rule[];
+
+        /** IDs of the rules to remove.
+         * Any invalid IDs will be ignored.
+         */
+        removeRuleIds?: number[];
+    }
+
+    export interface UpdateRulesetOptions {
+        /** The set of ids corresponding to a static Ruleset that should be disabled. */
+        disableRulesetIds?: string[];
+
+        /** The set of ids corresponding to a static Ruleset that should be enabled. */
+        enableRulesetIds?: string[];
+    }
+
+    export interface MatchedRuleInfoDebug {
+        /** Details about the request for which the rule was matched. */
+        request: RequestDetails;
+
+        rule: MatchedRule;
+    }
+
+    export interface Ruleset {
+        /** Whether the ruleset is enabled by default. */
+        enabled: boolean;
+
+        /** A non-empty string uniquely identifying the ruleset.
+         * IDs beginning with '_' are reserved for internal use.
+         */
+        id: string;
+
+        /** The path of the JSON ruleset relative to the extension directory. */
+        path: string;
+    }
+
+    export interface RulesMatchedDetails {
+        /** Rules matching the given filter. */
+        rulesMatchedInfo: MatchedRuleInfo[]
+    }
+
+    /** Returns the number of static rules an extension can enable before the global static rule limit is reached. */
+    export function getAvailableStaticRuleCount(callback: (count: number) => void): void;
+
+    /** Returns the current set of dynamic rules for the extension.
+     *
+     * @param callback Called with the set of dynamic rules.
+     * An error might be raised in case of transient internal errors.
+     */
+    export function getDynamicRules(callback: (rules: Rule[]) => void): void;
+
+    /** Returns the ids for the current set of enabled static rulesets.
+     *
+     * @param callback Called with a list of ids, where each id corresponds to an enabled static Ruleset. */
+    export function getEnabledRulesets(callback: (rulesetIds: string[]) => void): void;
+
+    /** Returns all rules matched for the extension.
+     * Callers can optionally filter the list of matched rules by specifying a filter.
+     * This method is only available to extensions with the declarativeNetRequestFeedback permission or having the activeTab permission granted for the tabId specified in filter.
+     * Note: Rules not associated with an active document that were matched more than five minutes ago will not be returned.
+     *
+     * @param filter An object to filter the list of matched rules.
+     * @param callback Called once the list of matched rules has been fetched.
+     * In case of an error, runtime.lastError will be set and no rules will be returned.
+     * This can happen for multiple reasons, such as insufficient permissions, or exceeding the quota.
+     */
+    export function getMatchedRules(filter: MatchedRulesFilter | undefined, callback: (details: RulesMatchedDetails) => void): void;
```

Comment:
I think `filter ?: MatchedRulesFilter` over `| undefined` for optional fields? Otherwise we need to explicitly pass undefined, instead of being able to just pass in one argument.

Reply:
Ah okay that makes sense, merging as is sounds good.

Reply:
done. will wait for the checks to conclude and merge it afterwards.

Reply:
whoops, now it needs another owner approval. :-)